### PR TITLE
Paragraph classes

### PR DESCRIPTION
**Pull request by adamdelaney**
Provide a classes selection field that can be implemented on paragraph bundles.

### CODE REVIEW

File: modules/uiowa_paragraphs/uiowa_paragraphs.module [hunk 1]
```diff
@@ -58,6 +58,14 @@ function uiowa_paragraphs_preprocess_paragraph(&$variables) {
   $admin_context = \Drupal::service('router.admin_context');
   if (!$admin_context->isAdminRoute()) {
     $paragraph = $variables['paragraph'];
+    if ($paragraph->hasField('field_uip_classes') && !$paragraph->get('field_uip_classes')->isEmpty()) {
+      $classes = [];
+      $values = $paragraph->get('field_uip_classes')->getValue();
+      array_walk_recursive($values, function ($v) use (&$classes) {
+        $classes['value'] = $v;
```
Comment (richardbporter) [inline modules/uiowa_paragraphs/uiowa_paragraphs.module]: This is reassigning `$classes['value']` for each iteration so only one class ever prints.

Comment (richardbporter) [reply modules/uiowa_paragraphs/uiowa_paragraphs.module]: I'm not sure what structure is required here to merge with `$variables['attributes']['class']` below.

**Comment by richardbporter** [comment]
Flattening the array seems to work. Tested the alter hook as well - just requires a key be set since that is the value used. It might be nice if the value was used instead but not a big deal.

**Comment by adamdelaney** [comment]
@richardbporter good catch and thanks for the commits fixing those issues. I'd prefer setting the key to the class so that the option value can be used for something more descriptive or human readable. For example, if we use [bootstrap utility classes](https://getbootstrap.com/docs/4.1/utilities/spacing/) like `mt3`we could provide a label like `Margin Top Medium`.

**Comment by richardbporter** [comment]
Ah, that makes sense. Ship it. 👍

**Pull request merged**